### PR TITLE
HW02: CUB: Mark Ipatov

### PR DESCRIPTION
**Pull request by MarkTheHopeful**
HW02 is done

### CODE REVIEW

File: Main.hs [hunk 1]
```diff
@@ -4,19 +4,29 @@ module Main where
 
 import Text.Printf (printf)
 import Control.Monad (unless)
-import Data.List (sort)
+import Data.List (sort, partition)
 
 quickSort :: [Int] -> [Int]
-quickSort = id 
+quickSort [] = []
+quickSort (x:xs) = (quickSort less) ++ [x] ++ (quickSort greaterOrEqual)
```
Comment (kajigor) [inline Main.hs:11]: It's better not to concatenate to a singleton list, use list constructor `:` instead

File: Main.hs [hunk 2]
```diff
@@ -4,19 +4,29 @@ module Main where
...
 
 map' :: (a -> b) -> [a] -> [b]
-map' = undefined 
+map' _ [] = []
+map' f (x:xs) = f x : map' f xs
 
 concatMap' :: (a -> [b]) -> [a] -> [b]
-concatMap' = undefined 
+concatMap' _ [] = []
+concatMap' f (x:xs) = (f x) ++ concatMap' f xs -- You can make it O(n), if you are ok with reversed order of each subarray
```
[the remark this answers is not in the document]
Comment (MarkTheHopeful) [reply Main.hs]: That makes sense, I removed the comment here. Still, I guess I've come up with a way to do this in O(n).

File: Main.hs (hunk 1 above)
Comment (MarkTheHopeful) [reply Main.hs]: Ok, there is an overhead on creating a list here, I agree. Fixed

Comment (kajigor) [reply Main.hs]: The overhead is not only because of creating list but because of traversing the `quickSort less` twice.

**Pull request closed without merging**